purchases/src/main/kotlin/com/revenuecat/purchases/Purchases.kt

```diff
@@ -382,47 +395,52 @@ class Purchases @VisibleForTesting(otherwise = VisibleForTesting.PRIVATE) intern
         message = "The listener has changed to accept a null Purchase on the onCompleted",
         replaceWith = ReplaceWith(
             expression = """
-                Purchases.sharedInstance.purchasePackage(activity, skuDetails, upgradeInfo, ProductChangeListener)
+                Purchases.sharedInstance.purchaseProduct(activity, skuDetails, upgradeInfo, ProductChangeListener)
             """
-        ),
-        level = DeprecationLevel.WARNING
+        )
     )
     fun purchaseProduct(
         activity: Activity,
         skuDetails: SkuDetails,
         upgradeInfo: UpgradeInfo,
         listener: MakePurchaseListener
     ) {
-        startProductChange(
+        purchaseProduct(
             activity,
             skuDetails.toProductDetails(),
-            null,
             upgradeInfo,
-            listener.toProductChangeListener()
+            listener.toProductChangeCallback()
```

Comment:
I think we should be using Callback rather than Listener, so I named the newer interfaces with ...Callback. I can change it if you think it's weird